docs/user-guide/federation/federated-ingress.md

```diff
@@ -0,0 +1,267 @@
+be healthy shards and will not direct traffic to any of these clusters.
+
+## Adding backend services and pods
+
+To render the underlying ingress shards healthy, we need to add
+backend Pods behind the service upon which the Ingress is based.  There are several ways to achieve this, but
+the easiest is to create a [Federated Service](federated-services.md) and 
+[Federated Replicaset](federated-replicasets.md).  Details of how those
+work are covered in the aforementioned user guides - here we'll simply use them, to
+create appropriately labelled pods and services in the 13 underlying clusters of
+our federation:
+
+``` shell
+kubectl --context=federation-cluster create -f services/nginx.yaml
+```
+
+``` shell
+  kubectl --context=federation-cluster create -f myreplicaset.yaml
+```
+
+You can verify this by checking in each of the underlying clusters, for example:
+
+``` shell
+kubectl --context=gce-asia-east1a get services nginx
+NAME      CLUSTER-IP     EXTERNAL-IP      PORT(S)   AGE
+nginx     10.63.250.98   104.199.136.89   80/TCP    9m
+```
+
+
+## Hybrid cloud capabilities
+
+Federations of Kubernetes Clusters can include clusters running in
+different cloud providers (e.g. Google Cloud, AWS), and on-premises
+(e.g. on OpenStack).  However, in Kubernetes v1.4, Federated Ingress is only
+supported across Google Cloud clusters.  In future versions we intend
+to support hybrid cloud Ingress-based deployments.
+
+## Discovering a federated ingress
+
+Ingress objects (in both plain Kubernets clusters, and in federations
+of clusters) expose one or more IP addresses (via
+the Status.Loadbalancer.Ingress field) that remains static for the lifetime
+of the Ingress object (in future, automatically managed DNS names
+might also be added).  All clients (whether internal to your cluster,
+or on the external network or internet) should connect to one of these IP
+or DNS addresses.  As mentioned above, all client requests are automatically
+routed, via the shortest network path, to a healthy pod in the
+closest cluster to the origin of the request. So for example, HTTP(S)
+requests from internet
+users in Europe will be routed directly to the closest cluster in
+Europe that has available capacity.  If there are no such clusters in
+Europe, the request will be routed to the next closest cluster
+(typically in the U.S.).
+
+## Handling failures of backend pods and whole clusters
+
+Ingresses are backed by Services, which are typically (but not always)
+backed by one or more ReplicaSets.  For Federated Ingresses, it is
+common practise to use the federated variants of Services and
+ReplicaSets (see [Federated Services](federated-services.md) and
+[Federated ReplicaSets](federated-services.md)) for this purpose, as
+described above.
+
+In particular, Federated ReplicaSets ensure that the desired number of
+pods are kept running in each cluster, even in the event of node
+failures.  And in the event of entire cluster or availability zone
```

Comment:
Remove "And"; just start with "In the event that the entire cluster or availability zone fails,"